.github/workflows/compile-kfp.yml

```diff
@@ -0,0 +1,87 @@
+name: Compile KFP pipelines
+
+on:
+  pull_request:
+    types: [opened, synchronize, reopened]
+    paths:
+      - "kubeflow-pipelines/docling-standard/**"
+      - "kubeflow-pipelines/docling-vlm/**"
+      - ".github/workflows/compile-kfp.yml"
+      - "tools/compare_yaml.py"
+  push:
+    branches: [ main ]
+    paths:
+      - "kubeflow-pipelines/docling-standard/**"
+      - "kubeflow-pipelines/docling-vlm/**"
+      - ".github/workflows/compile-kfp.yml"
+      - "tools/compare_yaml.py"
+
+permissions:
+  contents: read
+
+jobs:
+  compile:
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        include:
+          - name: docling-standard
+            dir: kubeflow-pipelines/docling-standard
+            cmd: python standard_convert_pipeline.py
+            out: standard_convert_pipeline_compiled.yaml
+          - name: docling-vlm
+            dir: kubeflow-pipelines/docling-vlm
+            cmd: python vlm_convert_pipeline.py
+            out: vlm_convert_pipeline_compiled.yaml
+
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+
+      - name: Set up Python 3.12
+        uses: actions/setup-python@v5
+        with:
+          python-version: "3.12"
+
+      # dependency for the compare script
+      - name: Install compare deps
+        run: |
+          python -m pip install --upgrade pip
+          pip install pyyaml
+
+      # snapshot the committed IR BEFORE compiling so we can diff later
+      - name: Snapshot baseline (${{ matrix.name }})
+        id: snap
+        run: |
+          set -euo pipefail
+          mkdir -p "$RUNNER_TEMP/baselines"
+          BASE="${{ matrix.dir }}/${{ matrix.out }}"
+          SNAP="$RUNNER_TEMP/baselines/${{ matrix.name }}.yaml"
+          if [ -f "$BASE" ]; then
+            cp "$BASE" "$SNAP"
+          else
+            : > "$SNAP"  # empty baseline if none exists yet
+          fi
+          echo "snap=$SNAP" >> "$GITHUB_OUTPUT"
+
+      # mirror local: install each pipeline's own requirements
+      - name: Install requirements for ${{ matrix.name }}
+        working-directory: ${{ matrix.dir }}
+        run: |
+          python -m pip install --upgrade pip
+          pip install -r requirements.txt
+
+      - name: Compile ${{ matrix.name }}
+        working-directory: ${{ matrix.dir }}
+        run: |
+          set -euxo pipefail
```

Comment:
I'd never seen the `set` command run like this in a github action so I did a little bit of reading. According to [the docs](https://docs.github.com/en/actions/reference/workflows-and-actions/workflow-syntax?versionId=free-pro-team%40latest&productId=actions&restPage=reference%2Cworkflows-and-actions%2Cworkflow-commands#exit-codes-and-error-action-preference) `set -e` is on by default and `set -o pipefail` in on by default for `bash`, which is the default shell on Ubuntu.

If we want the additional debugging it looks like `set -ux` would be all that's necessary.